xgcm/grid.py

```diff
@@ -868,6 +868,20 @@ def transform(
                 "`transform` can only be used on axes that are non-periodic. Pass `periodic=False` to `xgcm.Grid`."
             )
 
+        # complain if the target values are not provided as xr.dataarray
+        for var_name, variable, allowed_types in [
+            ("da", da, [xr.DataArray]),
+            ("target", target, [xr.DataArray, np.ndarray]),
+            ("target_data", target_data, [xr.DataArray]),
+        ]:
+            if not (
+                any([isinstance(variable, ty) for ty in allowed_types])
+                or variable is None
+            ):
+                raise ValueError(
+                    f"`{var_name}` needs to be a {' or '.join([str(a) for a in allowed_types])}. Found {type(variable)}"
```

Comment:
This will result in somewhat long error messages like 
```
`da` needs to be a <class 'xarray.core.dataarray.DataArray'>. Found <class 'xarray.core.dataset.Dataset'>"
```
@andersy005 is there a nice way to shorten these strings? Might not be necessary though...

Reply:
Try `variable.__class__.__name__` instead of `type(variable)`

```python
In [8]: ds.__class__
Out[8]: xarray.core.dataset.Dataset

In [9]: ds.__class__.__name__
Out[9]: 'Dataset'

In [10]: type(ds)
Out[10]: xarray.core.dataset.Dataset

```